### ISSUE TITLE
navigationDestination(selected: ) doesn't work on iOS 16.1 beta.

### ISSUE THREAD

**Issue by hmuronaka** [hmuronaka/NavigationDestinationSelectedViewModifier#1]
Despite being in the NavigationStack, Xcode displays the following message when this method is called.

> A `navigationDestination(isPresented:content:)` is outside an explicit NavigationStack, but inside the detail column of a NavigationSplitView, so it attempts to target the next column. There is no next column after the detail column.
>
>Did you mean to put the navigationDestination inside a NavigationStack?

**Comment by hmuronaka** [comment]
It seems that [navigationDestination(isPresented:destination:)](https://developer.apple.com/documentation/swiftui/view/navigationdestination(ispresented:destination:)) is broken or is changed on iOS 16.1 beta (20B5045d).
The following simple code doesn't work on iOS 16.1 beta.

I posted https://developers.apple.com/forums/thread/715589

# Sample Code

```swift
@main
struct StackViewSampleApp: App {
    var body: some Scene {
        WindowGroup {
            NavigationStack {
                ContentView()
            }
        }
    }
}

struct ContentView: View {
    var body: some View {
        List {
            NavigationLink("SubView1") {
                SubView1()
            }
        }
    }
}

struct SubView1: View {
    
    @State private var isPresent: Bool = false
     
    var body: some View {
        Button("A") {
            isPresent = true
        }
        .navigationDestination(isPresented: $isPresent) {
            Text("TEST")
        }
    }
}
```

**Comment by hmuronaka** [comment]
It seems the issue is fixed with Xcode 14.1 beta 3 (14B5033e) / iOS 16.1 beta(20B5056e)
